automation_tools/__init__.py

```diff
@@ -1465,3 +1469,178 @@ def setenforce(mode):
         ))
 
     run('setenforce {0}'.format(mode))
+
+
+def get_hostname_from_ip(ip):
+    """Retrives the hostname by logging into remote machine by IP.
+    Specially for the systems who doesnt support reverse DNS.
+    e.g usersys machines.
+
+    :param str ip: Remote host's IP address to retrive hostname.
+    """
```

Comment:
Add a line of whitespace between the last bit of prose and the closing `"""`. Also, don't list `str` as the type. Sphinx will try to link to the second argument when there's three arguments, and given that `str` is not defined anywhere in autmation-tools, the link will fail.

    :param ip: A string. The IP address of the remote host.

    """

And yes, I am guilty of listing un-linkable types as the middle argument in many docstrings. Forgive me. I taught myself Sphinx on-the-fly.